File: sukh_1684/problems/valid_parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        
        //'(', ')', '{', '}', '[' and ']'     
        Stack<Character> brackets = new Stack<>();
        for(int i=0;i<s.length();i++){
            char temp = s.charAt(i);
            if(temp == '(' || temp == '{' || temp == '['){
                brackets.push(temp);
            }
            else{
                if(brackets.isEmpty()){
                    return false;
                }
            char check = brackets.pop();

            if(temp == ')' && check!='('){
                return false;
            }
           if(temp == ']' && check!='['){
                return false;
            }
            if(temp == '}' && check!='{'){
                return false;
            }
            }
        }
            
         if(brackets.isEmpty())  {
        return true;
         }
        return false;
    }
}

File: sukh_1684/problems/valid_parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pointer = 0
        n = len(s)
        while (pointer < n):
            temp = s[pointer]
            if temp in ['(','{','[']:
                stack.append(temp)
            else:
                if not stack:
                    return False
                e = stack.pop()
                if temp == ')' and e != '(':
                    return False
                elif temp == '}' and e != '{':
                    return False
                elif temp == ']' and e != '[':
                    return False
                
            pointer += 1
        
        return (not stack)

        
            
            